src/main/react/app/containers/App/excludedDate/excludedDate.js

```diff
@@ -0,0 +1,210 @@
+        this.setState(this.getDatePresets()[this.state.selectedPreset]);
+        break;
+      case DATE_TYPE.TYPE_CHINESE_LUNAR:
+        let preset = this.getDatePresets()[this.state.selectedPreset];
+        this.setState(preset);
+
+        /*For types that are not gregorian, we calculate its occurrence
+        * using its corresponding algorithm.*/
+        this.setState({
+          nextOccurrence: nextOccurrenceChineseLunar(preset.params)
+        });
+    }
+  };
+
+  handleRepeatChange = (event) => {
+    this.setState({ repeat: event.target.checked });
+  };
+
+  handleRepeatPeriodChange = (e) => {
+    this.setState({ repeatPeriod: Number.parseInt(e.target.value) });
+  };
+
+  handleNoEndChange = (event) => {
+    this.setState({ noEnd: event.target.checked });
+  };
+
+  handleIntervalChange = (e) => {
+    this.setState({ repeatInterval: e.target.value });
+  };
+
+  handleCountChange = (e) => {
+    if (e.target.value === "-1") {
+      this.setState({ noEnd: true });
+    }
+    this.setState({ repeatCount: e.target.value });
+  };
+
+  handlePresetChange = (e) => {
+    this.setState({
+      /*Set the preset index for later apply.*/
+      selectedPreset: parseInt(e.target.value)
+    });
+  };
+
+  handleDateTypeChange = (e) => {
+    this.setState({
+      selectedDateType: e.target.value,
+      /*Set the default index of the selected preset*/
+      selectedPreset: 0
+    });
+  };
+
+
+
+
+  /*For the selected date type, get its presets.*/
+  getDatePresets() {
+    return getDatePresets(this.state.selectedDateType);
+  }
+
+  /*Tell the parent that this child want to be toggle open state.*/
+  toggleEdit = () => {
+    /*Call onEdit, also emit data to parent(for serializing use).*/
+    this.props.onEdit(this.props.index, !this.props.opened, this.state);
+  };
+
+  deleteDate = () => {
+    if (window.confirm("Are you sure to delete?")
+    ) {
+      this.props.onDelete(this.props.index);
+    }
+  };
+
+  /*Helper function to judge whether the day is based on gregorian calendar*/
+  isGregorian = () => {
+    return this.state.type === DATE_TYPE.TYPE_GREGORIAN;
+  };
+
+  componentDidMount() {
+    this.setState(this.props.date,() => {
+      this.props.onEdit(this.props.index, true, this.state);
+    });
+  }
+
+  render() {
+    const { repeat, noEnd } = this.state;
+    return (
+      <div className={"config-item"}>
+        {/*Allow each date item to open or close, need help of the parent component.*/}
+        {this.props.opened ? <div>
+            {NameInput.call(this)}
+            {TimezoneInput.call(this)}
+            {PresetSelect.call(this)}
+            {RepeatCheckbox.call(this)}
+            {repeat && <div>
+              {RepeatPeriod.call(this)}
+              {RepeatInterval.call(this)}
+              {RepeatCount.call(this)}
+            </div>}
+            {this.isGregorian() && DateInput.call(this,
+              {
+                field: "startDate",
+                name: "Start Date"
```

Comment:
We don't have to address it right now, but do we have a solution for making strings localizable?